### DISCUSSION TITLE
Use undici instead of http Agent

### DISCUSSION THREAD

**Discussion by olexiyb** [ClickHouse/clickhouse-js#152] [Ideas]
I propose to take a look into [undici](https://undici.nodejs.org/#/

[Why Undici](https://nodejs.medium.com/introducing-undici-4-1e321243e007)

**Comment by slvrtrn** [comment]
As of 0.2.x, we can utilize `@clickhouse/client-common` to create a new Node.js client implementation alongside the existing one (similarly to `@clickhouse/client-web`).

I guess "socket hang up" and, generally, how KeepAlive works are the main issues with the current version that uses just the standard library. 

So, the main question is: is it possible to create a barebones example where undici will not fail with these "socket hang up" errors where the current implementation will? Cause the worst-case scenario will be that we'll have the same errors, but now with two different flavors of the library, which I would like to avoid.

**Reply by peter-leonov-ch** [reply]
I can see two immediate benefits of using the native `fetch`:
* the connection polling has been progressing rapidly since `http.Agent` and likely does not have a racy socket manager anymore
* fetch API as a whole is highly composable with `Request` separated from the actual transport

It also promises good support for promises :)
